src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/P2PProtocolHandler.cs

```diff
@@ -109,7 +109,7 @@ public override void HandleMessage(Packet msg)
                     HandleHello(Deserialize<HelloMessage>(msg.Data));
                     
                     foreach (Capability capability in
-                        _agreedCapabilities.GroupBy(c => c.ProtocolCode).Select(c => c.OrderBy(v => v.Version).Last()))
+                        _agreedCapabilities.GroupBy(c => c.ProtocolCode).Select(c => c.OrderBy(v => v.Version).Last()).OrderBy(c => c.ProtocolCode))
```

Comment:
Maybe move _agreedCapabilities and _availableCapabilities from List to SortedSet ?

Reply:
I can do it, but I wonder if it has sense. With lists we are ordering only once, when handling Hello. With sorted set we will have comparison every time when adding capability to _agreedCapabilities, so with Eth, Aa and let's say Snap we will have two comparisons instead of one ordering when using lists. And in _availableCapabilities order doesn't matter at all.